pkg/driver/node.go

```diff
@@ -99,19 +108,27 @@ func (d *Driver) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolu
 		mountpointArgs = compileMountOptions(mountpointArgs, mountFlags)
 	}
 
-	//Checking if the target directory is already mounted with a volume.
-	mounted, err := d.isMounted(target)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "Could not check if %q is mounted: %v", target, err)
+	klog.V(4).Infof("NodePublishVolume: mounting %s at %s with options %v", bucket, target, mountpointArgs)
+	hostPluginDirEnv := os.Getenv(hostPluginDirEnv)
+	if hostPluginDirEnv == "" {
+		// set the default in case the env variable isn't found
+		hostPluginDirEnv = "/var/lib/kubelet/plugins/s3.csi.aws.com/"
 	}
-	if !mounted {
-		klog.V(4).Infof("NodePublishVolume: mounting %s at %s with options %v", bucket, target, mountpointArgs)
-		if err := d.Mounter.Mount(bucket, target, fstype, mountpointArgs); err != nil {
-			os.Remove(target)
-			return nil, status.Errorf(codes.Internal, "Could not mount %q at %q: %v", bucket, target, err)
-		}
-		klog.V(4).Infof("NodePublishVolume: %s was mounted", target)
+	hostTokenPath := path.Join(hostPluginDirEnv, "token")
```

Comment:
do we need to do any validation that `hostPluginDirEnv` ends in `/` or is that safe to assume?

Reply:
`path.Join` takes care of that for you. It handles either one.